File: 0056-Merge-Intervals/lc0056.js
// JavaScript solution.

/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    if (!intervals.length) { return []; }
    intervals.sort((a,b) => a[0] - b[0]);
    let ans = [];
    for (let pair of intervals) {
        if (ans.length > 0 && pair[0] <= ans[ans.length-1][1]) {
            ans[ans.length-1][1] = Math.max(ans[ans.length-1][1], pair[1]);
        } else {
            ans.push([...pair]);
        }
    }
    return ans;
};

File: 0056-Merge-Intervals/lc0056.py
# Python solution.

class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        ans = []
        for pair in sorted(intervals, key=lambda x: x[0]):
            if ans and pair[0] <= ans[-1][1]:
                ans[-1][1] = max(ans[-1][1], pair[1])
            else:
                ans.append(pair[:])
        return ans
